Mathlib/Analysis/Normed/Group/SeparationQuotient.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2024 Yoh Tanimoto. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yoh Tanimoto
+-/
+import Mathlib.Analysis.Normed.Group.Uniform
+import Mathlib.Analysis.Normed.MulAction
+
+/-!
+# The null subgroup in a seminormed commutative group
+
+For any `SeminormedAddCommGroup M`, the quotient `SeparationQuotient M` by the null subgroup is
+defined as a `NormedAddCommGroup` instance in `Mathlib.Analysis.Normed.Group.Uniform`. Here we
+define the null space as a subgroup.
+
+## Main definitions
+
+We use `M` to denote seminormed groups.
+All the following definitions are in the `SeparationQuotient` namespace. Hence we can access
+`SeparationQuotient.normedMk` as `normedMk`.
+
+* `nullSubgroup` : the subgroup of elements `x` with `‖x‖ = 0`.
+
+If `E` is a vector space over `𝕜` with an appropriate continuous action, we also define the null
+subspace as a submodule of `E`.
+
+* `nullSubmodule` : the subspace of elements `x` with `‖x‖ = 0`.
+
+## Implementation details
+
+For any `SeminormedAddCommGroup M`, we define a norm on `SeparationQuotient M` by
```

Comment:
This is no longer in this file, right? I think we can just drop this section.

Reply:
dropped